### PR TITLE
CBG-3916 Use DCP feed to purge documents in viewsAndGSIBucketReadier

### PR DESCRIPTION
**Pull request by adamcfraser**
Uses one-shot DCP feed to purge tombstones instead of all docs index.

CBG-3916

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [x] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/2408/

### CODE REVIEW

File: db/util_testing.go [hunk 1]
```diff
@@ -218,6 +218,94 @@ func EmptyPrimaryIndex(ctx context.Context, dataStore sgbucket.DataStore) error
...
+		}
+	case <-timeout:
+		return 0, fmt.Errorf("timeout waiting for purge DCP feed to complete")
+	}
+
+	tbp.Logf(ctx, "Finished purge DCP feed ... Total docs purged: %d", purgedDocCount)
+	return purgedDocCount, nil
+}
+
 // emptyAllDocsIndex ensures the AllDocs index for the given bucket is empty, including tombstones which aren't found when emptying the primary index.
```
Comment (bbrks) [inline db/util_testing.go:316]:
```suggestion
// emptyAllDocsIndex ensures the AllDocs index for the given bucket is empty, including tombstones which aren't found when emptying the primary index.
// nolint:unused
```

if we still want to keep it around

File: db/util_testing.go [hunk 2]
```diff
@@ -218,6 +218,94 @@ func EmptyPrimaryIndex(ctx context.Context, dataStore sgbucket.DataStore) error
...
+	case err := <-doneChan:
+		if err != nil {
+			tbp.Logf(ctx, "purgeDCPFeed finished with error: %v", err)
+		}
+	case <-timeout:
+		return 0, fmt.Errorf("timeout waiting for purge DCP feed to complete")
+	}
+
+	tbp.Logf(ctx, "Finished purge DCP feed ... Total docs purged: %d", purgedDocCount)
+	return purgedDocCount, nil
```
Comment (bbrks) [inline db/util_testing.go:313]: `purgeErrors` isn't being used - previous implementation had `ErrorOrNil()` at the final return to bubble up to test framework.

```suggestion
	return purgedDocCount, purgeErrors.ErrorOrNil()
```

**Review by bbrks — CHANGES_REQUESTED** [review]

**Comment by bbrks** [comment]
Should we merge this into `main` and rebase `release/anemone`? I think we'll still need this to correctly clean up docs using the eventing xattr.

**Comment by adamcfraser** [comment]
Moved to PR on main: https://github.com/couchbase/sync_gateway/pull/6817

**Pull request closed without merging**